# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/25 10:10:51+00:00 aia21@cantab.net 
#   NTFS: Add printk rate limiting for ntfs_warning() and ntfs_error() when
#         compiled without debug.  This avoids a possible denial of service
#         attack.  Thanks to Carl-Daniel Hailfinger from SuSE for pointing this
#         out.
# 
# fs/ntfs/ChangeLog
#   2005/01/25 10:10:42+00:00 aia21@cantab.net +7 -0
#   Update
# 
# fs/ntfs/debug.c
#   2005/01/25 10:10:42+00:00 aia21@cantab.net +8 -0
#   Add printk rate limiting for ntfs_warning() and ntfs_error() when
#   compiled without debug.  This avoids a possible denial of service
#   attack.  Thanks to Carl-Daniel Hailfinger from SuSE for pointing this out.
# 
diff -Nru a/fs/ntfs/ChangeLog b/fs/ntfs/ChangeLog
--- a/fs/ntfs/ChangeLog	2005-01-25 03:12:55 -08:00
+++ b/fs/ntfs/ChangeLog	2005-01-25 03:12:55 -08:00
@@ -25,6 +25,13 @@
 	- Enable the code for setting the NT4 compatibility flag when we start
 	  making NTFS 1.2 specific modifications.
 
+2.1.23-WIP
+
+	- Add printk rate limiting for ntfs_warning() and ntfs_error() when
+	  compiled without debug.  This avoids a possible denial of service
+	  attack.  Thanks to Carl-Daniel Hailfinger from SuSE for pointing this
+	  out.
+
 2.1.22 - Many bug and race fixes and error handling improvements.
 
 	- Improve error handling in fs/ntfs/inode.c::ntfs_truncate().
diff -Nru a/fs/ntfs/debug.c b/fs/ntfs/debug.c
--- a/fs/ntfs/debug.c	2005-01-25 03:12:55 -08:00
+++ b/fs/ntfs/debug.c	2005-01-25 03:12:55 -08:00
@@ -53,6 +53,10 @@
 	va_list args;
 	int flen = 0;
 
+#ifndef DEBUG
+	if (!printk_ratelimit())
+		return;
+#endif
 	if (function)
 		flen = strlen(function);
 	spin_lock(&err_buf_lock);
@@ -93,6 +97,10 @@
 	va_list args;
 	int flen = 0;
 
+#ifndef DEBUG
+	if (!printk_ratelimit())
+		return;
+#endif
 	if (function)
 		flen = strlen(function);
 	spin_lock(&err_buf_lock);
